### PR TITLE
Sb 16706 Draft: validated users with respect to state

### PR DESCRIPTION
**Pull request by Hari-stackroute**
This pr generates 2 more files:
validated-user-summary.json: Two fields were added "subOrgRegistered", "rootOrgRegistered"
`{"districts":2,"blocks":2,"schools":2,"subOrgRegistered":3,"registered":5,"rootOrgRegistered":2}`

validated-user-detail-state.csv: Validated user details wrto state

Sl|District name|District ext. ID|Block name|Block ext. ID|School name|School ext. ID|Teacher name|Teacher ext. ID|Teacher Diksha ID
--|-------------|----------------|----------|-------------|-----------|--------------|------------|---------------|-----------------
1|||||APSchool|orgext1|emp4Name|apext1|

### CODE REVIEW

File: batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportHelper.scala [hunk 1]
```diff
@@ -30,16 +30,17 @@ trait StateAdminReportHelper extends  BaseReportsJob {
       .withColumn("explodedlocation", explode(when(size(col("locationids")).equalTo(0), array(lit(null).cast("string")))
         .otherwise(when(col("locationids").isNotNull, col("locationids"))
           .otherwise(array(lit(null).cast("string"))))))
-
     val subOrgJoinedDF = subOrgDF
-      .where(col("status").equalTo(1) && not(col("isrootorg")))
+      .where(col("status").equalTo(1))
```
Comment (indrajra) [inline batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportHelper.scala]: if you have removed condition, can you rename var? It is misleading.

File: batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportJob.scala [hunk 2]
```diff
@@ -86,34 +93,38 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
+        val districtUserResult = userDistrictSummaryDF.groupBy(col("slug"), col("District name").as("districtName")).
+            agg(countDistinct("Block id").as("blocks"),countDistinct(claimedShadowUserDF.col("orgextid")).as("schools"), count("userextid").as("registered"))
+        saveUserDistrictSummary(districtUserResult)
+        
+        districtUserResult
     }
 
     def saveValidatedUserDetailsReport(userDistrictSummaryDF: DataFrame, url: String) {
-      val window = Window.partitionBy("slug").orderBy(asc("District name"))
+      val window = Window.partitionBy("slug").orderBy(desc("District name"))
```
Comment (indrajra) [inline batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportJob.scala]: Shouldn't this be reverted?

File: batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportHelper.scala (hunk 1 above)
Comment (Hari-stackroute) [reply batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportHelper.scala]: changed to activeSuborgDF

File: batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportJob.scala (hunk 2 above)
Comment (Hari-stackroute) [reply batch-models/src/main/scala/org/ekstep/analytics/job/report/StateAdminReportJob.scala]: Yes this should be asc order.

**Pull request merged**